Source/WTF/wtf/text/AtomStringImpl.cpp

```diff
@@ -476,7 +476,7 @@ RefPtr<AtomStringImpl> AtomStringImpl::lookUpSlowCase(StringImpl& string)
 
 RefPtr<AtomStringImpl> AtomStringImpl::add(std::span<const char8_t> characters)
 {
-    HashedUTF8Characters buffer { characters, computeUTF16LengthWithHash(spanReinterpretCast<const char8_t>(characters)) };
```

Comment:
Seems like we should consider revising `spanReinterpretCast` to fail to compile if it's redundant. Could easily do it by adding a `static_assert`.

Reply:
Oh, nice.

Reply:
Filed:  Bug [281994](https://bugs.webkit.org/show_bug.cgi?id=281994): Add static_assert to spanReinterpretCast<>() to prevent unnecessary casts

Reply:
Posted WebKit PR <https://github.com/WebKit/WebKit/pull/35638>